ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Background.hs

```diff
@@ -532,5 +533,8 @@ addBlockRunner
   => ChainDbEnv m blk
   -> m Void
 addBlockRunner cdb@CDB{..} = forever $ do
-    blockToAdd <- getBlockToAdd cdbBlocksToAdd
-    addBlockSync cdb blockToAdd
+    let trace = traceWith cdbTracer . TraceAddBlockEvent
+    trace $ PoppedBlockFromQueue Heralding
```

Comment:
We could have a combinator a bit richer than `enclosingWith` that expects the passed (inner) action to also return the argument to `Heralded`. But I figure it's not worth the extra layer of indirection.

Reply:
Yeah, if this comes up more often, we can add functions like these (with better names, hopefully):

```haskell
encloseWith' ::
     Monad m
  => Tracer m (Enclosing' e)
  -> m e
  -> m ()
encloseWith' tracer action =
    encloseWith'' tracer ((,()) <$> action)

encloseWith'' ::
     Monad m
  => Tracer m (Enclosing' e)
  -> m (e, a)
  -> m a
encloseWith'' tracer action = do
    traceWith tracer RisingEdge
    (e, a) <- action
    traceWith tracer $ FallingEdge' e
    pure a
```